dusty/commands/upgrade.py

```diff
@@ -0,0 +1,70 @@
+import json
+import os
+import shutil
+from subprocess import check_output, check_call, CalledProcessError
+import urllib
+
+from .. import constants
+from ..daemon import close_client_connection
+from ..log import log_to_client
+from ..payload import daemon_command
+
+
+def _get_latest_version():
+    """Gets latest Dusty binary version using the GitHub api"""
+    url = 'https://api.github.com/repos/gamechanger/dusty/releases/latest'
+    conn = urllib.urlopen(url)
+    json_data = conn.read()
+    return json.loads(json_data)['tag_name']
+
+def _get_binary_url(version):
+    return 'https://github.com/gamechanger/dusty/releases/download/{}/dusty'.format(version)
+
+def _get_binary_location():
+    return check_output(['which', 'dusty']).rstrip()
```

Comment:
I think you can use the proc module we have elsewhere to figure out the binary location of the current process. This which check is insufficient and could fail based on PATH settings